repository/src/main/java/org/apache/atlas/repository/store/graph/v2/ClassificationAssociator.java

```diff
@@ -160,6 +187,53 @@ public void setClassifications(Map<String, AtlasEntityHeader> map) throws AtlasB
                     throw e;
                 }
             }
+
+            //send Notifications & update __classificationText
+            RequestContext.get().clearEntityCache();
+
+            AtlasPerfMetrics.MetricRecorder recorder = RequestContext.get().startMetricRecord("commitChanges.notify");
+            Map<AtlasClassification, Collection<Object>> deleted = RequestContext.get().getDeletedClassificationAndVertices();
+            Set<AtlasVertex> allVertices = new HashSet<>();
+
+            if (MapUtils.isNotEmpty(deleted)) {
+                for (AtlasClassification deletedClassification: deleted.keySet()) {
+                    Collection<Object> vertices =  deleted.get(deletedClassification);
+                    List<AtlasEntity> propagatedEntities = new ArrayList<>();
+
+                    for (Object obj: vertices) {
+                        AtlasVertex vertex = (AtlasVertex) obj;
+                        AtlasEntity entity = instanceConverter.getAndCacheEntity(GraphHelper.getGuid(vertex), IGNORE_REL);
+
+                        allVertices.add(vertex);
+                        propagatedEntities.add(entity);
+                    }
+
+                    entityChangeNotifier.onClassificationsDeletedFromEntities(propagatedEntities, Collections.singletonList(deletedClassification));
+                }
+            }
+
+            Map<AtlasClassification, Collection<Object>> added = RequestContext.get().getAddedClassificationAndVertices();
+            if (MapUtils.isNotEmpty(added)) {
+                for (AtlasClassification addedClassification: added.keySet()) {
+                    Collection<Object> vertices =  added.get(addedClassification);
+                    List<AtlasEntity> propagatedEntities = new ArrayList<>();
+
+                    for (Object obj: vertices) {
+                        AtlasVertex vertex = (AtlasVertex) obj;
+                        AtlasEntity entity = instanceConverter.getAndCacheEntity(GraphHelper.getGuid(vertex), IGNORE_REL);
+
+                        allVertices.add(vertex);
+                        propagatedEntities.add(entity);
+                    }
+
+                    entityChangeNotifier.onClassificationsAddedToEntities(propagatedEntities, Collections.singletonList(addedClassification), false);
+                }
+            }
+            entityGraphMapper.updateClassificationText(null, allVertices);
+            transactionInterceptHelper.intercept();
```

Comment:
We will be commiting updated classifications in `entitiesStore.updateClassifications(entityGuid, list);`,

Can updating this part later will break the atomicity of the request or it will be committed at last?

Reply:
It will be comited at last once for all assets there in request, also it will only commit __classificationText as other writes are already comitted in add/update or delete methods

Reply:
So it is two step write? First all the assets then at last we will commit classification text